onnx/checker.cc

```diff
@@ -221,6 +225,31 @@ void check_tensor(const TensorProto& tensor, const CheckerContext& ctx) {
 #undef check_field
 }
 
+void check_sparse_tensor(
+    const SparseTensorProto& sparse,
+    const CheckerContext& ctx) {
+  int64_t nnz = 0;
+  if (sparse.has_values()) {
+    const TensorProto& values = sparse.values();
+    check_tensor(values, ctx);
+    // TBD: Should we generalize and permit tensors of rank > 1?
+    if (values.dims().size() != 1)
+      fail_check("Sparse tensor values must have rank 1.");
+    nnz = values.dims(0);
+  }
+  int64_t num_indices = 0;
+  if (sparse.has_indices()) {
```

Comment:
Should we throw if sparse tensor has no indices? Also, would you think checking if those indices are properly bound by `dims`  is one thing we need to do here?

Reply:
Added the checks.